nifi-sysunite-processors/src/test/java/com/sysunite/nifi/VirtuosoClientTest.java

```diff
@@ -0,0 +1,70 @@
+package com.sysunite.nifi;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.Before;
+import org.junit.Test;
+import virtuoso.jena.driver.ISQLChannel;
+import virtuoso.jena.driver.Virtuoso;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+public class VirtuosoClientTest {
+
+  private TestRunner testRunner;
+
+  /**
+   *
+   *  -> Run the docker tenforce/virtuoso:1.0.0-virtuoso7.2.2 image before running this test
+   *  -> Set properties in config.properties
+   *
+   */
+
+
+
+  @Before
+  public void init() {
+
+    Virtuoso quadStore = new Virtuoso("jdbc:virtuoso://"+TestConfig.getVirtuosoAddress(), TestConfig.getVirtuosoUser(), TestConfig.getVirtuosoPassword());
+    ISQLChannel.sendQuery(quadStore.getVirtGraph(), "DELETE FROM DB.DBA.RDF_QUAD");
+
+    ISQLChannel.sendQuery(quadStore.getVirtGraph(), "sparql INSERT DATA INTO GRAPH <x> {  <http://ba#a> <b> \"dev\" . } ");
+    ISQLChannel.sendQuery(quadStore.getVirtGraph(), "sparql INSERT DATA INTO GRAPH <y> {  <c> <d> <e> . } ");
+
+
+    testRunner = TestRunners.newTestRunner(VirtuosoClient.class);
+  }
+
+  @Test
+  public void testOnTrigger() throws IOException {
```

Comment:
what happens if there are no results (eg empty db), no nullpointerexceptions somehwere? might be worth to test that scenario too